code/controllers/subsystem/ai_controllers.dm

```diff
@@ -37,6 +53,15 @@ SUBSYSTEM_DEF(ai_controllers)
 		ai_controller.SelectBehaviors(wait * 0.1)
 		if(!LAZYLEN(ai_controller.current_behaviors)) //Still no plan
 			COOLDOWN_START(ai_controller, failed_planning_cooldown, AI_FAILED_PLANNING_COOLDOWN)
+		if(ai_controller.can_idle)
+			var/found_interesting = FALSE
+			for(var/client/client_found in GLOB.clients)
+				if(get_dist(get_turf(client_found.mob), get_turf(ai_controller.pawn)) <= ai_controller.interesting_dist)
+					found_interesting = TRUE
+					break
```

Comment:
for the record this is a little insane when the spatial grid exists, but it's at least better